main.rb

```diff
@@ -0,0 +1,232 @@
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr_ret = []
+    i = 0
+    length.times do
+      arr_ret.push(self[i]) if yield(self[i])
+      i += 1
+    end
+    arr_ret
+  end
+
+  def my_all?(par = nil)
+    ret_value = true
+
+    if block_given?
+      i = 0
+      length.times do
+        return false unless yield(self[i])
+
+        i += 1
+      end
+    elsif !par
+      i = 0
+      length.times do
+        return false unless self[i]
+
+        i += 1
+      end
+    else
+      if par.is_a? Regexp
+        my_each { |x| ret_value = false unless x =~ par }
+        return ret_value
+      end
+      if par.is_a? Class
+        my_each { |x| ret_value = false unless x.is_a?(par) }
+        return ret_value
+      end
+      unless par.nil?
+        my_each { |x| ret_value = false unless x == par }
+      end
+    end
+    ret_value
+  end
+
+  def my_any?(par = nil)
+    ret_value = false
+    if block_given?
+      i = 0
+      length.times do
+        return true if yield(self[i])
+
+        i += 1
+      end
+    elsif !par
+      i = 0
+      length.times do
+        return true if self[i]
+
+        i += 1
+      end
+    else
+      if par.is_a? Regexp
+        my_each { |x| ret_value = true if x =~ par }
+        return ret_value
+      end
+      if par.is_a? Class
+        my_each { |x| ret_value = true if x.is_a?(par) }
+        return ret_value
+      end
+      unless par.nil?
+        my_each { |x| ret_value = true if x == par }
+      end
+    end
+    ret_value
+  end
+
+  def my_none?(par = nil)
+    ret_value = true
+    if block_given?
+      i = 0
+      length.times do
+        return false if yield(self[i])
+
+        i += 1
+      end
+    elsif !par
+      i = 0
+      length.times do
+        return false if self[i]
+
+        i += 1
+      end
+    else
+      if par.is_a? Regexp
+        my_each { |x| ret_value = false if x =~ par }
+        return ret_value
+      end
+      if par.is_a? Class
+        my_each { |x| ret_value = false if x.is_a?(par) }
+        return ret_value
+      end
+      unless par.nil?
+        my_each { |x| ret_value = false if x == par }
+      end
+    end
+    ret_value
+  end
+
+  def my_count(par = nil)
+    counter = 0
+    if block_given?
+      i = 0
+      length.times do
+        counter += 1 unless yield(self[i]).nil?
+        i += 1
+      end
+    elsif !par
+      return length
+    else
+      if par.is_a? Regexp
+        my_each { |x| counter += 1 if x =~ par }
+        return counter
+      end
+      if par.is_a? Class
+        my_each { |x| counter += 1 if x.is_a?(par) }
+        return counter
+      end
+      unless par.nil?
+        my_each { |x| counter += 1 if x == par }
+      end
+    end
+    counter
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    arr_ret = []
+    i = 0
+    length.times do
+      arr_ret.push(yield(self[i]))
+      i += 1
+    end
+    arr_ret
+  end
```

Comment:
### [OPTIONAL]

* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_map 
- [ ] in the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so I strongly recommend working on it

Given 
```ruby
my_proc = Proc.new {|num| num > 10 }
```
Result
```ruby
puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  => true true false false
```

Reply:
There were two "my_map" and my_map_proc the second one was the one that accepts proc as an argument

Reply:
Now I changed the names the original my_map it's now my_map_old and "my_map_proc" it's "my_map"